src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt

```diff
@@ -50,24 +48,49 @@ class BookingControllerMockMvcTests @Autowired constructor(
     }
 
     /**
+     * Fail to get a booking
+     */
+    @Test
+    fun getNonexistentBookingTest() {
+        // arrange
+        // act
+        val result = mockMvc.perform(MockMvcRequestBuilders.get("/v1/booking/999"))
+
+        // assert
+        result.andExpect(MockMvcResultMatchers.status().isNotFound)
+    }
+
+    /**
+     * Fail to get a booking
+     */
+    @Test
+    fun getMalformedURITest() {
+        // arrange
+        // act
+        val result = mockMvc.perform(MockMvcRequestBuilders.get("/v1/booking/notanumber"))
+
+        // assert
+        result.andExpect(MockMvcResultMatchers.status().is4xxClientError)
+    }
+
+    /**
      * Create a booking
      */
     @Test
     fun createBookingTest() {
         // arrange
-        val startDateTime = ZonedDateTime.parse("2017-09-26T09:00:00.000-04:00")
-        val endDateTime = ZonedDateTime.parse("2017-09-26T10:00:00.000-04:00")
+        val startDateTime = LocalDateTime.parse("2017-09-26T09:00:00")
+        val endDateTime = LocalDateTime.parse("2017-09-26T10:00:00")
         val request = BookingRequest(1, "New Meeting", startDateTime, endDateTime)
 
         // act
         val result = mockMvc.perform(MockMvcRequestBuilders.post("/v1/booking")
             .contentType(MediaType.APPLICATION_JSON)
             .content(mapper.writeValueAsString(request)))
 
+        println(MockMvcResultMatchers.content())
         // assert
         result.andExpect(MockMvcResultMatchers.status().`is`(HttpStatus.CREATED.value()))
         result.andExpect(MockMvcResultMatchers.jsonPath<String>("$.subject", Matchers.equalToIgnoringCase("New Meeting")))
-        result.andExpect(MockMvcResultMatchers.jsonPath<BigDecimal>("$.startDateTime", Matchers.equalTo(BigDecimal("1506430800.000000000"))))
-        result.andExpect(MockMvcResultMatchers.jsonPath<BigDecimal>("$.endDateTime", Matchers.equalTo(BigDecimal("1506434400.000000000"))))
     }
```

Comment:
might wanted to add some additional tests around validation of datetime?

Reply:
I want to add these back.. having some issues with the default serializer

Reply:
I just mean more tests for when the client sends an invalid date/time